common/metrics/metrics.go

```diff
@@ -0,0 +1,125 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package metrics
+
+import (
+	"time"
+
+	"golang.org/x/exp/event"
+)
+
+// Mostly cribbed from
+// https://github.com/temporalio/sdk-go/blob/master/internal/common/metrics/handler.go
+// and adapted to depend on golang.org/x/exp/event
+type (
+	// MetricProvider represents the main dependency for instrumentation
+	MetricProvider interface {
+		// WithTags creates a new MetricProvder with provided []Tag
+		// Tags registered with the resulting MetricProvider are only the Tags provided
+		// Tags are not merged with registered Tags from the source MetricProvider
```

Comment:
Is this the right behavior? I thought we merge tags when creating child scope with existing code. Could you help verify this is the same behavior as existing one? If we change the behavior, we might lose tags.

Reply:
I will verify this and make changes to not break the current behavior.

Reply:
its a bit convoluted to tell, but merging is the current behavior.
going to work that in and make sure the comments are accurate.